src/main/java/com/openmrs/migrator/core/services/impl/BootstrapServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package com.openmrs.migrator.core.services.impl;
+
+import com.openmrs.migrator.core.services.BootstrapService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BootstrapServiceImpl implements BootstrapService {
+
+  private static Logger log = LoggerFactory.getLogger(BootstrapServiceImpl.class);
+
+  int numberOfCreatedFiles = 0;
+
+  private final Path TARGET_JOB_FOLDER = Paths.get("pdiresources/jobs/");
+
+  private final Path TARGET_TRANSFORMATION_FOLDER = Paths.get("pdiresources/transformations/");
+
+  /**
+   * Creates folders structure, if a one of the folder exists a warn log is raised informing that
+   * the folder or won't be created because it already exists
+   *
+   * @return number of created files
+   */
+  @Override
+  public int createDirectoryStructure(List<String> dirList, Path settingsProperties)
+      throws IOException {
+
+    log.info("Starting creating folder structure");
+
+    dirList.forEach(
+        x -> {
+          if (!checkIfPathExist(Paths.get(x))) {
+            try {
+              Files.createDirectory(Paths.get(x));
+              log.info("Folder: " + x + " created sucessfully");
+              if (x.equals("pdiresources/")) {
+                Files.createDirectory(TARGET_JOB_FOLDER);
+                Files.createDirectory(TARGET_TRANSFORMATION_FOLDER);
+              }
+              numberOfCreatedFiles++;
```

Comment:
Is there much benefit in us tracking how many folders were created?

Reply:
re-factored:  it return a stream of directories,  more precise to test now